src/libxtp/threecenter_gwbse.cc

```diff
@@ -283,7 +285,7 @@ void TCMatrix_gwbse::FillBlockCUDA(
   return;
 }
 
-std::pair<CudaMatrix, CudaMatrix> TCMatrix_gwbse::SendDFTMatricesToGPU(
+std::vector<CudaMatrix> TCMatrix_gwbse::SendDFTMatricesToGPU(
```

Comment:
Ahh okay, a) you can still return a std::array and b) I would split that function up into two `CopyDFTMatricestoGPU` and `CreateIntermediateMatrices` or sth similar because it is not clear why SendMatricesToGPU returns so many. I see why it is nice because you get the indices for free, but you can get some of the indeces from the membervariabels and the rest you just pass as ints.

Reply:
done